File: 02 tests/ex9_regexp.js
// run with: deno run ex9_regexp.js | node ex9_regexp.js

/* ex9_regexp.js

Extract digits from a 10^7 random string

:author:  wolf
:created: 2025.05.22
*/

const iterations = 3;
const timing_results = [];

// define the test function
// --- function to be timed-- -

function jsRegex(n) {
    const chars = "abcdefghijklmnopqrstuvwxyz0123456789";
    const text = Array.from({ length: n }, () => chars[Math.floor(Math.random() * chars.length)]).join('');
    return (text.match(/\d/g) || []).length;
}


for (let i = 0; i < iterations; i++) {
    const start = performance.now();

    // call the test function
    jsRegex(10**7);

    const end = performance.now();

    timing_results.push(end - start);
}


console.log("Elapsed time: ",
    timing_results.reduce((a, b) => a + b) / iterations / 1000, "seconds");

File: 02 tests/ex9_regexp.py
#!/usr/bin/env python3

""" ex9_regexp.py

Extract digits from a 10^7 random string

:author:	wolf
:created:	2025.05.22
"""

import time
import string
import random
import re

iterations = 3
timing_results = []

# define the test function
# --- function to be timed ---

def python_regex(n):
    text = ''.join(random.choice(string.ascii_lowercase + string.digits)
                   for _ in range(n))
    return len(re.findall(r'\d', text))


for i in range(iterations):
    start = time.perf_counter()

    # call the test function
    python_regex(10**7)

    end = time.perf_counter()

    timing_results.append(end - start)


print(f"Elapsed time:  {sum(timing_results)/iterations:.16f} seconds")
